### PR TITLE
libfyaml 0.7.12 (new formula)

### PR DESCRIPTION
**Pull request by eisenlohr**
A fancy 1.2 YAML and JSON parser/writer.
Fully feature complete YAML parser and emitter, supporting the latest YAML spec and passing the full YAML testsuite.

- [ y ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ y ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ y ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ y ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ y ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ y ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
./configure: line 19414: syntax error near unexpected token `LIBYAML,'
  ./configure: line 19414: `PKG_CHECK_MODULES(LIBYAML,  yaml-0.1 , HAVE_LIBYAML=1, HAVE_LIBYAML=0)'

File: Formula/libfyaml.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! For Homebrew/homebrew-core
+    # this will need to be a test that verifies the functionality of the
+    # software. Run the test with `brew test libfyaml`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+
```
Comment (chenrui333) [inline Formula/libfyaml.rb:30]:
```suggestion
```

File: Formula/libfyaml.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+class Libfyaml < Formula
+  desc "Fully feature complete YAML parser and emitter"
+  homepage "https://github.com/pantoniou/libfyaml"
+  url "https://github.com/pantoniou/libfyaml/archive/refs/tags/v0.7.12.tar.gz"
+  sha256 "864e60dc1863e2ba2048b9765d9addd653f6c387e835438f09cb5f26a2a352e1"
```
Comment (cho-m) [inline Formula/libfyaml.rb:5]: The official source tarball is probably better as it includes the generated configure script, which is our standard way of building stable autotools-based packages:
```suggestion
  url "https://github.com/pantoniou/libfyaml/releases/download/v0.7.12/libfyaml-0.7.12.tar.gz"
  sha256 "485342c6920e9fdc2addfe75e5c3e0381793f18b339ab7393c1b6edf78bf8ca8"
```

File: Formula/libfyaml.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  license "MIT"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "gcc" => :build
+  depends_on "libtool" => :build
```
Comment (cho-m) [inline Formula/libfyaml.rb:11]: `autoconf`, `automake`, and `libtool` are usually for generating configure, so are probably unnecessary with other source tarball. `gcc` should not be needed and will not be used in current state as Homebrew's shims will select Apple Clang on macOS and GCC 11 on Linux.
```suggestion
```

File: Formula/libfyaml.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+class Libfyaml < Formula
+  desc "Fully feature complete YAML parser and emitter"
+  homepage "https://github.com/pantoniou/libfyaml"
+  url "https://github.com/pantoniou/libfyaml/releases/download/v0.7.12/libfyaml-0.7.12.tar.gz"
+  sha256 "485342c6920e9fdc2addfe75e5c3e0381793f18b339ab7393c1b6edf78bf8ca8"
+  license "MIT"
+
+  def install
+    system "./bootstrap.sh"
```
Comment (SMillerDev) [inline Formula/libfyaml.rb:9]: This isn't needed since the configure already exists
```suggestion
```

**Comment by eisenlohr** [comment]
I am not clear on how to instruct the build to not use a flat namespace, which appears to cause the present error...

**Comment by carlocab** [comment]
> @carlocab do you know if that fix made it into a autotools release yet?

Yep, fixed in the latest Libtool

**Comment by SMillerDev** [comment]
> could you be more specific what you refer to by "that fix"? I am, unfortunately, not very experienced in adding a formula to homebrew.

It was a question for someone else. And it's now answered.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Just some audit failure with flat_namespace patch, LGTM

**Comment by chenrui333** [comment]
Thanks folks!

**Pull request closed without merging**